runtime/gc_base/ContinuationObjectList.cpp

```diff
@@ -46,13 +46,22 @@ MM_ContinuationObjectList::MM_ContinuationObjectList()
 }
 
 MM_ContinuationObjectList *
-MM_ContinuationObjectList::newInstanceArray(MM_EnvironmentBase *env, uintptr_t arrayElements)
+MM_ContinuationObjectList::newInstanceArray(MM_EnvironmentBase *env, uintptr_t arrayElements, MM_ContinuationObjectList *listsToCopy, uintptr_t listsToCopySize)
 {
 	MM_ContinuationObjectList *continuationObjectLists;
 
 	continuationObjectLists = (MM_ContinuationObjectList *)env->getForge()->allocate(sizeof(MM_ContinuationObjectList) * arrayElements,  MM_AllocationCategory::FIXED, J9_GET_CALLSITE());
 	if (NULL != continuationObjectLists) {
-		for (uintptr_t index = 0; index < arrayElements; index++) {
+		Assert_MM_true(arrayElements >= listsToCopySize);
+		/* Check whether a new array instance in being created from an existing array. If so, copy over the elements first. */
+		if (listsToCopySize > 0) {
+			for (uintptr_t index = 0; index < listsToCopySize; index++) {
+				continuationObjectLists[index] = listsToCopy[index];
```

Comment:
Formally, we are still constructing continuationObjectLists[index], but just from another object, so it's more correct to rely on copy constructor, although in practice it may not make any difference.
You may give it a try, although I'm not sure if copy constructor is callable with placement:

```new(&continuationObjectLists[index]) MM_ContinuationObjectList(listsToCopy[index]);```

Reply:
Because `continuationObjectLists[index]` is uninited, I believe this assignment implicitly makes use of the copy ctor and not the assignment operator:
`continuationObjectLists[index] = listsToCopy[index];`

Reply:
Personally I think the assignment operator approach looks more clean than placement

Reply:
actually, not. this relies on 'operator =', and there are small diffs between the 2 (lots of refs on the Net), which may be immaterial here

Reply:
Ultimately decided to leave as is - discussed over IM